provider/kubernetes/kubernetes.go

```diff
@@ -312,6 +314,15 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 							protocol = "https"
 						}
 
+						switch getStringValue(i.Annotations, annotationKubernetesProtocol, protocol) {
+						case allowedProtocolHTTPS:
+							protocol = allowedProtocolHTTPS
+						case allowedProtocolH2C:
+							protocol = allowedProtocolH2C
+						default:
```

Comment:
I'm not too much of a fan of falling back to some default value for the case where the user entered invalid data. How about indicating an error and skipping the offending Ingress if the given annotation value is unsupported?

That's also how we seem to handle things in most other cases of the Kubernetes provider.